### PR TITLE
Added New label-related commands

### PR DESCRIPTION
**Pull request by antonovchtd**
There are three new commands:
• unlabel <id> - asks for label and removed it from task with ID <id>
• UNLABEL <id> - clears all labels of task with ID <id>
• labels <id> - shows string of labels of task with ID <id>

### CODE REVIEW

File: src/ui/actions/ClearAllLabelsOfTaskAction.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+#ifndef TASKMANAGER_SRC_UI_ACTIONS_CLEARALLLABELSOFTASKSACTION_H_
+#define TASKMANAGER_SRC_UI_ACTIONS_CLEARALLLABELSOFTASKSACTION_H_
+
+#include "Action.h"
+
+class ClearAllLabelsOfTaskAction : public Action {
+public:
+    ClearAllLabelsOfTaskAction(const Core::TaskID &);
```
Comment (AntonApraksin) [inline src/ui/actions/ClearAllLabelsOfTaskAction.h]: Mark explicit

File: src/ui/steps/ClearAllLabelsStep.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#define TASKMANAGER_SRC_UI_STEPS_CLEARALLLABELSSTEP_H_
+
+
+#include "Step.h"
+#include "utilities/StepUtils.h"
+#include "ui/actions/ClearAllLabelsOfTaskAction.h"
+
+class ClearAllLabelsStep : public Step {
+public:
+    using Step::Step;
```
Comment (AntonApraksin) [inline src/ui/steps/ClearAllLabelsStep.h]: Is there a state in the interface `Step`?

File: src/ui/steps/ShowAllLabelsStep.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+//
+// Created by Anton Ovcharenko on 18.01.2022.
+//
+
+#include "ShowAllLabelsStep.h"
+#include "ui/Context.h"
+
+std::unique_ptr<Action> ShowAllLabelsStep::genAction(Context &context) {
+    auto labels = context.tasks()[0].data().labels();
```
Comment (AntonApraksin) [inline src/ui/steps/ShowAllLabelsStep.cpp]: Unchecked access

**Review by AntonApraksin — CHANGES_REQUESTED** [review]

File: src/ui/actions/ClearAllLabelsOfTaskAction.h (hunk 1 above)
Comment (antonovchtd) [reply src/ui/actions/ClearAllLabelsOfTaskAction.h]: fixed everywhere

File: src/ui/steps/ClearAllLabelsStep.h (hunk 2 above)
Comment (antonovchtd) [reply src/ui/steps/ClearAllLabelsStep.h]: yes, will add a layer and fix in another PR

File: src/ui/steps/ShowAllLabelsStep.cpp (hunk 3 above)
Comment (antonovchtd) [reply src/ui/steps/ShowAllLabelsStep.cpp]: should not be a problem, but added a check nevertheless for good practice

File: src/ui/actions/GetTaskToShowLabelsAction.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class GetTaskToShowLabelsAction : public Action {
+public:
+    explicit GetTaskToShowLabelsAction(const std::string &arg);
+
+public:
+    ActionResult execute(const std::shared_ptr<ModelInterface> &) override;
+
+private:
+    std::string arg_;
```
Comment (KharchenkoMaks) [inline src/ui/actions/GetTaskToShowLabelsAction.h:18]: better to give TaskId to Actions, not human readable strings

File: test/model/TaskManagerTest.cpp [hunk 5]
```diff
@@ -299,6 +299,84 @@ TEST_F(TaskManagerTest, shouldAddMultipleLabels){
...
+    t.set_due_date(time(nullptr));
+    t.set_is_complete(false);
+    Core::TaskID id = *tm.Add(t).id;
+    std::string label = "testing";
+    std::string label2 = "testing2";
+    tm.AddLabel(id, label);
+    tm.AddLabel(id, label2);
+    tm.ClearLabel(id, label);
+    auto tasks = tm.getTasks();
+    EXPECT_EQ(1, tasks[0].data().labels().size());
```
Comment (KharchenkoMaks) [inline test/model/TaskManagerTest.cpp]: better to use ASSERT_EQ when checking container size to fail test before out of range exception

**Review by KharchenkoMaks — CHANGES_REQUESTED** [review]

File: test/model/TaskManagerTest.cpp (hunk 5 above)
Comment (antonovchtd) [reply test/model/TaskManagerTest.cpp]: done

File: src/ui/actions/GetTaskToShowLabelsAction.h (hunk 4 above)
Comment (antonovchtd) [reply src/ui/actions/GetTaskToShowLabelsAction.h:18]: will be done in a refactoring PR

File: test/model/TaskManagerTest.cpp [hunk 6]
```diff
@@ -299,6 +299,84 @@ TEST_F(TaskManagerTest, shouldAddMultipleLabels){
...
+    t.set_priority(p);
+    t.set_due_date(time(nullptr));
+    t.set_is_complete(false);
+    Core::TaskID id = *tm.Add(t).id;
+    std::string label = "testing";
+    std::string label2 = "testing2";
+    tm.AddLabel(id, label);
+    tm.AddLabel(id, label2);
+    tm.ClearLabels(id);
+    EXPECT_TRUE(tm.getTasks()[0].data().labels().empty());
```
Comment (KharchenkoMaks) [inline test/model/TaskManagerTest.cpp:377]: better to check vector size

**Review by KharchenkoMaks — APPROVED** [review]

File: src/model/TaskManager.h [hunk 7]
```diff
@@ -33,6 +33,8 @@ class TaskManager : public ModelInterface {
     ActionResult Delete(const Core::TaskID &id, bool deleteChildren) override;
     ActionResult Validate(const Core::TaskID &id) const override;
     ActionResult AddLabel(const Core::TaskID &, const std::string &) override;
+    ActionResult ClearLabel(const Core::TaskID &, const std::string &) override;
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.h:36]: `RemoveLabel` would be a better name

File: src/model/help.txt [hunk 8]
```diff
@@ -23,6 +23,11 @@
     label <id>          prompts the user to enter a label to be assigned to the task with ID
                         <id>.
 
+    unlabel <id>        prompts the user to enter a label to be removed from the task with ID
+                        <id>.
+
+    UNLABEL <id>        clears all labels for the task with ID <id>.
```
Comment (vladimir-moiseiev) [inline src/model/help.txt:29]: avoid un* verbs.

File: src/ui/actions/GetTaskToShowLabelsAction.cpp [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+// Created by Anton Ovcharenko on 18.01.2022.
+//
+
+#include "GetTaskToShowLabelsAction.h"
+
+GetTaskToShowLabelsAction::GetTaskToShowLabelsAction(const std::string &arg) : arg_{arg} {
+}
+
+ActionResult GetTaskToShowLabelsAction::execute(const std::shared_ptr<ModelInterface> &model) {
+    std::optional<Core::TaskID> id{Core::TaskID()};
```
Comment (vladimir-moiseiev) [inline src/ui/actions/GetTaskToShowLabelsAction.cpp:11]: you don't need optional when it's a input parameter.

**Review by vladimir-moiseiev — APPROVED** [review]
please check my comments

File: test/model/TaskManagerTest.cpp (hunk 6 above)
Comment (antonovchtd) [reply test/model/TaskManagerTest.cpp:377]: noted

**Pull request merged**